### PR TITLE
Remove property validation from explain admin

### LINKED ISSUE #7314: Stop using internal property in explainAdminCommand
## Feature Request

In the ExplainAdminCommand we're using internal doctrine properties:
https://github.com/sonata-project/SonataAdminBundle/blob/master/src/Command/ExplainAdminCommand.php#L138
https://github.com/sonata-project/SonataAdminBundle/blob/master/src/Command/ExplainAdminCommand.php#L157
We should stop using it.

Imho we should just remove from the explanation the properties constraint and the getter constraints.
The command is made to explain an admin, not the entity. And since the validation group is not provided in the explanation, it's not really useful to list all the constraint.

WDYT @sonata-project/contributors ?

### PR DESCRIPTION
**Pull request by VincentLanglet**
## Subject

I am targeting this branch, because BC-break.

Closes #7314.

## Changelog

```markdown
### Removed
- Properties and getters validation from ExplainAdminCommand.
```

### CODE REVIEW

File: src/Command/ExplainAdminCommand.php [hunk 1]
```diff
@@ -33,15 +31,9 @@ final class ExplainAdminCommand extends Command
      */
     private $pool;
 
-    /**
-     * @var MetadataFactoryInterface
-     */
-    private $validator;
-
-    public function __construct(Pool $pool, MetadataFactoryInterface $validator)
+    public function __construct(Pool $pool)
```
Comment (OskarStark) [inline src/Command/ExplainAdminCommand.php:34]: Shall we make this class `@internal` to remove it from the BC promise?

Comment (VincentLanglet) [reply src/Command/ExplainAdminCommand.php:37]: Done for all the command and twig extensions.

File: src/Command/ExplainAdminCommand.php [hunk 2]
```diff
@@ -34,14 +32,11 @@ final class ExplainAdminCommand extends Command
     private $pool;
 
     /**
-     * @var MetadataFactoryInterface
+     * @internal This class should only be used through the console
```
Comment (dmaicher) [inline src/Command/ExplainAdminCommand.php:35]: should this annotation be on the constructor?

File: src/Command/ExplainAdminCommand.php (hunk 1 above)
Comment (dmaicher) [reply src/Command/ExplainAdminCommand.php:37]: > to me this class is not meant to be extended

the class is final already? So not sure I see the benefit here

Comment (VincentLanglet) [reply src/Command/ExplainAdminCommand.php:37]: I understand it for the constructor signature

File: src/Command/ExplainAdminCommand.php (hunk 2 above)
Comment (VincentLanglet) [reply src/Command/ExplainAdminCommand.php:35]: Imho only the constructor should be internal. Adding this to the whole would say we can change the behavior of the command

**Review by dmaicher — APPROVED** [review]

**Review by core23 — APPROVED** [review]

**Pull request merged**